main.tf

```diff
@@ -0,0 +1,176 @@
+}
+
+resource "azurerm_log_analytics_solution" "logs" {
+  solution_name         = "ContainerInsights"
+  location              = azurerm_log_analytics_workspace.logs.location
+  resource_group_name   = azurerm_resource_group.k8s.name
+  workspace_resource_id = azurerm_log_analytics_workspace.logs.id
+  workspace_name        = azurerm_log_analytics_workspace.logs.name
+
+  plan {
+    publisher = "Microsoft"
+    product   = "OMSGallery/ContainerInsights"
+  }
+}
+
+resource "azuread_application" "app" {
+  name = local.cluster_name
+
+
+  app_role {
+    allowed_member_types = [
+      "User",
+      "Application",
+    ]
+
+    description  = "Admins can manage roles and perform all task actions"
+    display_name = "Admin"
+    is_enabled   = true
+    value        = "Admin"
+  }
+}
+
+resource "random_password" "aks" {
+  length  = 24
+  special = true
+}
+
+resource "azuread_service_principal" "sp" {
+  application_id = azuread_application.app.application_id
+
+  tags = [local.cluster_name]
+}
+
+resource "azuread_service_principal_password" "sp" {
+  service_principal_id = azuread_service_principal.sp.id
+  value                = random_password.aks.result
+  end_date             = "2099-01-01T01:00:00Z"
+}
+
+resource "azurerm_role_assignment" "sp" {
+  scope                = data.azurerm_subscription.current.id
+  role_definition_name = "Owner"
+  principal_id         = azuread_service_principal.sp.object_id
+}
+
+resource "azurerm_kubernetes_cluster" "aks" {
+  lifecycle {
+    ignore_changes = [
+      default_node_pool[0].node_count
+    ]
+  }
+
+  name                = local.cluster_name
+  location            = azurerm_resource_group.k8s.location
+  resource_group_name = azurerm_resource_group.k8s.name
+  dns_prefix          = local.cluster_name
+
+  default_node_pool {
+    name                = "platform"
+    node_count          = var.node_pools.platform.cluster_auto_scaling_max_count
+    vm_size             = var.node_pools.platform.vm_size
+    availability_zones  = var.node_pools.platform.zones
+    max_pods            = 250
+    os_disk_size_gb     = 128
+    node_taints         = var.node_pools.platform.taints
+    enable_auto_scaling = var.node_pools.platform.cluster_auto_scaling
+    min_count           = var.node_pools.platform.cluster_auto_scaling_min_count
+    max_count           = var.node_pools.platform.cluster_auto_scaling_max_count
+  }
+
+  service_principal {
+    client_id     = azuread_service_principal.sp.application_id
+    client_secret = azuread_service_principal_password.sp.value
+  }
+
+  addon_profile {
+    oms_agent {
+      enabled                    = true
+      log_analytics_workspace_id = azurerm_log_analytics_workspace.logs.id
+    }
+  }
+
+  network_profile {
+    load_balancer_sku  = "standard"
+    network_plugin     = "azure"
+    network_policy     = "calico"
+    dns_service_ip     = "10.0.0.10"
+    docker_bridge_cidr = "172.17.0.1/16"
+    service_cidr       = "10.0.0.0/16"
```

Comment:
Doesn't matter much since this is for internal testing for the foreseeable future, but might as well make this `100.97.0.0/16` to match other stuff we have / avoid conflicting with any connected private subnets.

Reply:
@Secretions  I assume you mean `10.97.0.0/16` - 100.X.X.X is not a private cidr block. Correct?

Reply:
No, `100.97.0.0/16` is correct. It's a trick I pulled from kops, which defaults to using `100.96.0.0/16` and `100.97.0.0/16` for pod and service cidrs respectively. They are in a special address space that is reserved for carrier-grade NAT, which makes it a useful range because it *is* a private, but one that you can safely assume isn't going to be used by another private network that you want to route to.

Reply:
Okay. Do you have a link handy where this is described?

Reply:
Here's some relevant stuff:

https://en.wikipedia.org/wiki/IPv4_shared_address_space
https://tools.ietf.org/html/rfc6598#section-7
https://en.wikipedia.org/wiki/Carrier-grade_NAT